src/NHibernate/Connection/ConnectionProvider.cs

```diff
@@ -112,9 +112,49 @@ protected virtual void ConfigureDriver(IDictionary<string, string> settings)
 				{
 					throw new HibernateException("Could not create the driver from " + driverClass + ".", e);
 				}
+
+#pragma warning disable 618
+				switch (driver)
+				{
+					case NpgsqlDriver _:
+						LogObsoleteDriver(nameof(NpgsqlDriver), "PostgreSql", "PostgreSqlDriver");
+						break;
+					case MySqlDataDriver _:
+						LogObsoleteDriver(nameof(MySqlDataDriver), "MySql", "MySqlDriver");
+						break;
+					case FirebirdClientDriver _:
+						LogObsoleteDriver(nameof(FirebirdClientDriver), "Firebird", "FirebirdDriver");
+						break;
+					case OracleManagedDataClientDriver _:
+						LogObsoleteDriver(nameof(OracleManagedDataClientDriver), "Oracle.Managed", "OracleManagedDriver");
+						break;
+					case SQLite20Driver _:
+						LogObsoleteDriver(nameof(SQLite20Driver), "SQLite", "SQLiteDriver");
+						break;
+					case Sql2008ClientDriver _:
+						LogObsoleteDriver(nameof(Sql2008ClientDriver), "SqlServer", "SqlServer2008Driver");
+						break;
+					case SqlClientDriver _:
+						LogObsoleteDriver(nameof(SqlClientDriver), "SqlServer", "SqlServer2000Driver");
+						break;
+					case SqlServerCeDriver _:
+						LogObsoleteDriver(nameof(SqlServerCeDriver), "SqlServer.Compact", "SqlServerCompactDriver");
+						break;
+				}
+#pragma warning restore 618
 			}
 		}
 
+		private static void LogObsoleteDriver(string obsoleteDriverName, string nugetName, string newDriverName)
+		{
+			log.Error(
```

Comment:
An obsolete is usually a warning, not an error. I think it should change to `Warn`.